### ISSUE TITLE
Compatibility with JurassiCraft's fossil blocks

### ISSUE THREAD

**Issue by HariboTer** [Aang23/UndergroundBiomesConstructs#45]
Hi, I'm currently trying to make this compatible with the fossil blocks from the JurassiCraft mod. The problem is that each of their fossil blocks has 4 texture variants, but its all the same block (seems to be done with blockstates / variants). I tried to achieve it by putting the texture variants on the block with 4 different entries with meta values from 0 to 3, but to no avail - all variants get replaced by the same single texture one :-(

Could you please take a look at that?

**Comment by Aang23** [comment]
Support will be added in the next release.

**Comment by HariboTer** [comment]
I just realized that there is another side to this which I haven't mentioned yet, which is that JurassiCraft also contains fossilized nest ores that have different textures on top than they have on the sides, which as far as I know is also not (yet) possible to add UBifying support for.

**Comment by Aang23** [comment]
Not that kind of ores are not yet supported...

**Comment by Aang23** [comment]
However blockstates and meta ore supported.

**Comment by HariboTer** [comment]
It appears to be done via blockstates, but even if so, I'd have no idea how to handle it (at least it's not said in the [guide](https://github.com/kyeiti/UndergroundBiomesConstructs/wiki/How-to-add-ore-support-for-mods).

Anyway, it's all the same block with the same meta, but it has a json file in the "blockstates" folder that reads
`{
  "variants": {
    "normal": [
      {
        "model": "jurassicraft:plant_fossil_block_0",
        "weight": 1
      },
      {
        "model": "jurassicraft:plant_fossil_block_1",
        "weight": 1
      },
      {
        "model": "jurassicraft:plant_fossil_block_2",
        "weight": 1
      },
      {
        "model": "jurassicraft:plant_fossil_block_3",
        "weight": 1
      }
    ]
  }
}
`

The models then are the usual blocks with a different texture each.

**Comment by Aang23** [comment]
Well that a standard blockstate / meta use case.
I just tried and it worked for me. I'll be adding native support for those who are standard ores in 1.3.2.

**Comment by HariboTer** [comment]
I'm not really following what you mean.
I tried giving myself one of JurassiCraft's plant fossil blocks, but any meta <= 0 gave me the normal plant fossil block and any meta >= 1 gave me a purple error textured block, which could however be placed like the normal plant fossil block.

So there is exactly one plant fossil block, but when placed it randomly chooses one of the 4 appearances.

I can get UBifying to replace them with a UBified plant fossil block, but then it is always the same one and the difference in appearance is lost.

If you got it working, could you upload the corresponding json file as an example? It could serve as a temporary guide for that case until the actual guide is updated.

**Comment by Aang23** [comment]
Oh I just understood what you mean. This is one block with different textures randomly choosen. (for the plant fossil).
This is not supported in UBC.

However the fossil_0 block behaves correctly, with standard metas. 
Note that a meta can only be >= 0.

As those are edge cases, and most mods don't do that, I'm surely gonna implement them directly, bypassing the Ore system.

**Comment by HariboTer** [comment]
Ah, now I understand. The dinosaur fossil do use meta to specify the species of dinosaur. It's nice to know that this can be looked up rather easily with console commands :-)

However, the dinosaur fossil blocks also do the same thing as the plant fossils with also 4 different textures (not related to the meta).

Will there be support to specify texture variants for a UBified block?

**Comment by Aang23** [comment]
Those 2 blocks are gonna get their own UBC variants.
I want to keep the ore integration system simple, and implementing such a thing would only be useful for some rare mods.
What that mean is that the meta-driven one will be supported trough the standard ore system, and those that need multi faces / are not meta-driven will be added the same way I did with Quark's Speleothems.

**Comment by HariboTer** [comment]
Yeah, it's a bit of a tough question. On the one hand, I can understand that hardcoding it can probably be done faster. On the other side, since those texture variants seem to be a standard Minecraft/Forge feature, I can see more mods using it, though I agree that using texture variants for ores is something you're more likely to see in resource packs than in mods themselves.

I'm kind of on the fence on this, because to me, the quark speleothems seem much more of a special case than multi-side or varying textures, but I'm also not familiar enough with the UBC code to estimate the effort required for both options. Personally however I'd be leaning towards adding it to the ore integration system, if nothing else then just because I'm always in favor of increasing end-user customizability^^

**Comment by Aang23** [comment]
I think I'm gonna implement in the ore system... But don't expect that to be done quickly thought.

**Comment by HariboTer** [comment]
Yes, I know I'm not entitled to anything here and will wait patiently :-)

Just gave my opinion because you literally asked for it ;-)

**Comment by HariboTer** [comment]
By the way, I also used my program to generate delta textures for the JurassiCraft blocks that should be UBified. I hope I'm not too late to spare you some work^^
[jurassicraft.zip](https://github.com/Aang23/UndergroundBiomesConstructs/files/2944201/jurassicraft.zip)

**Comment by Aang23** [comment]
No, not yet, thanks!
